plutarch-docs/Usage/Tracing.md

```diff
@@ -12,8 +12,26 @@ import Plutarch.Prelude ()
 
 # Tracing
 
-You can use the functions `ptrace`, `ptraceError`, `ptraceIfFalse`, `ptraceIfTrue` (from `Plutarch.Trace` or `Plutarch.Prelude`) for tracing. 
-These behave similarly to the ones you're used to from [PlutusTx](https://playground.plutus.iohkdev.io/doc/haddock/plutus-tx/html/PlutusTx-Trace.html).
+Fundamentally, there are two kinds of traces Plutarch can add to your code:
 
-If you have the `development` flag for `plutarch` turned on - you'll see the trace messages appear in the trace log during script evaluation. When not 
-in development mode - these functions basically do nothing.
+* Info tracing, which is the 'regular' kind of tracing; and
+* Debug tracing, which is supposed to be more verbose and provide more detail
+  for debugging purposes.
+
+The basic way you can add traces to your code is using `ptraceInfo` to add an
+info trace, and `ptraceDebug` to add a debug trace. `Plutarch.Trace` (and
+`Plutarch.Prelude`) export additional functions for more specific cases,
+including ones that include use of `PShow`, and that only inject a trace under
+certain conditions. See those modules, and their documentation, for more
+details.
+
+If you have the `development` flag for `plutarch` turned on - you'll see the 
+trace messages appear in the trace log during script evaluation. When not 
+in development mode, these functions basically do nothing.
+
+# Important note
+
+Use of `PShow` is strongly discouraged in any tracing code (and any code in
```

Comment:
But you wouldn't compile this when going on-chain?

Reply:
True. This might not be the best place for this warning.

Reply:
I think this note is in lightof https://github.com/Plutonomicon/plutarch-plutus/issues/697

If we correctly handle pruning of pshow according to the trace config, we don't need warning